lib/cfn-nag/custom_rules/password_base_rule.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require 'cfn-nag/util/enforce_reference_parameter'
+require 'cfn-nag/util/enforce_string_or_dynamic_reference'
+require_relative 'base'
+
+class PasswordBaseRule < BaseRule
+  def resource_type
+    raise 'must implement in subclass'
+  end
+
+  def password_property
+    raise 'must implement in subclass'
+  end
+
+  def audit_impl(cfn_model)
+    resources = cfn_model.resources_by_type(resource_type)
+
+    violating_resources = resources.select do |resource|
+      if resource.send(password_property).nil?
+        false
```

Comment:
it's probably a harmless "pass through" but the password property isn't optional for all of these resources.  it is optional for resources that can come from a snapshot, but the various directory resources require it

Reply:
May be a good idea to create a `password_required` that defaults to true that can be overridden in the implementing subclass